core/shared/src/main/scala/zio/FiberRefs.scala

```diff
@@ -56,15 +56,15 @@ final class FiberRefs private (
    * invocation of [[forkAs]] when we already know that the map will not be
    * transformed
    */
-  private var needsTransformWhenForked: Boolean = true
+  private[this] lazy val needsTransformWhenForked: Boolean =
```

Comment:
Not too happy that we're using lazy vals for this, but I wanted to avoid recomputing them all the time on `forkAs` as it happens every time we fork a FiberRef.

Alternatively, I could reimplement this using a var and a def, WDUT?

Reply:
`var` and `def` would be superior to `lazy val` which synchronizes.

Reply:
Agreed, changed the implementation to use `def`/`var` instead